File: mylibrary-backend/src/main/java/model/Emprestimo.java
package model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Emprestimo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "livro_id", nullable = false)
    private Livro livro;

    @Column(nullable = false)
    private String nomePessoa;

    private String telefone;

    @Column(nullable = false)
    private LocalDate dataEmprestimo;

    @Column(nullable = false)
    private LocalDate dataDevolucao;

    private LocalDate dataDevolucaoEfetiva;
}

File: mylibrary-frontend/src/app/models/Emprestimo.ts
export interface Emprestimo {
  id?: number;
  livroId?: number;
  tituloLivro?: string;
  livro?: {
    id: number;
  };
  nomePessoa: string;
  telefone: string;
  dataEmprestimo?: string;
  dataDevolucaoPrevista: string;
  dataDevolucaoEfetiva?: string;
}
